Add simple setup.py file

mdehavensift opened the pull request:
Added a simple setup.py file make it easy to install for use in other projects, like xplan-api.

There is a problem with this, though, as it does not specify the missing sbol dependency (see issue #3). I'm not sure where that is supposed to come from.

mwes commented:
@mdehavensift can do we some basic OS detection/python versioning here and install the necessary wheel if the OS is linux based? Perhaps something using sys.platform? win/macosx should work fine (pySBOL publishes these)

2.7:  https://github.com/tcmitchell/pySBOL/blob/ubuntu/Ubuntu_16.04_64_2/dist/pySBOL-2.3.0.post11-cp27-none-any.whl?raw=true
3.5:  https://github.com/tcmitchell/pySBOL/blob/ubuntu/Ubuntu_16.04_64_3/dist/pySBOL-2.3.0.post11-cp35-none-any.whl?raw=true
3.6:  https://github.com/tcmitchell/pySBOL/blob/ubuntu/Ubuntu_16.04_64_3/dist/pySBOL-2.3.0.post11-cp36-none-any.whl?raw=true

mdehavensift commented:
@mwes Yeah, I'll work on adding something in to grab those wheels

mdehavensift commented:
Reached out to Tom Mitchell to get instructions for how he made these wheel files. Current wheel files appear to only work on Ubuntu 16+. They don't work on my debian machine due to the gcc libraries that it came with.

If it's not too difficult, I'd like to make some more wheel files to expand the compatibility. Otherwise, a user would just need to manually upgrade their gcc versions, which may take a bit of time. Or work in an Ubuntu 16+ docker container.

mwes commented:
Agree on expanding compatibility with more wheels. Do you have what you need there? Tom more details, as you note.

mdehavensift commented:
Added ability to setup script to download and install custom pysbol wheel files for linux. Currently only supports Ubuntu. Plan to support more distributions soon.

mwes commented:
Thanks @mdehavensift! This has some mixed commits from Nic...hopefully that's OK. If not intended or ready, we can slice those out. Otherwise, this looks fine to me. @nroehner @tramyn can we merge this?

mdehavensift commented:
@nroehner @tramyn, Sorry I merged this branch before either of you weighed in. I reverted the merge and created a new request for this branch (can't reopen a pull request) if you want to comment on it, see #6

@mwes I think Nic's commits were only included because of a rebase or something like that. When I reverted the pull request, only my changes were reverted. So Nic's commits that were in this branch must have already been pushed into master before this pull request merged.

tramyn commented:
@mdehavensift No objections with the change here.
